### PR TITLE
Strict types for getActors method

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

More correct fix of #15900 for main branch to avoid bad SQL requests.

### CODE REVIEW

File: src/CommonITILActor.php [hunk 1]
```diff
@@ -92,16 +92,18 @@ public function isAttach2Valid(array &$input)
...
-    public function getActors($items_id)
+    public function getActors(int $items_id): array
     {
         /** @var \DBmysql $DB */
         global $DB;
 
         $users = [];
         $iterator = $DB->request([
-            'FROM'   => $this->getTable(),
+            'FROM'   => static::getTable(),
```
Comment (trasher) [inline src/CommonITILActor.php:106]: This should be changed on 10.0/bf branch

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -123,7 +123,7 @@ public function loadUsers(): void
     {
         if (!empty($this->userlinkclass) && isset($this->fields['id'])) {
             $class = new $this->userlinkclass();
-            $this->lazy_loaded_users = $class->getActors($this->fields['id']);
+            $this->lazy_loaded_users = $class->getActors((int) $this->fields['id']);
```
Comment (trasher) [inline src/CommonITILObject.php:126]: This (and other similar changes) should be changed on 10.0/bf branch

**Review by trasher — COMMENTED** [review]
There are at least some changes that really should be applied on bugfixes branch. I wonder if the whole PR should not target bugfixes branch, this does not seems to be a BC break.

File: src/CommonITILObject.php (hunk 2 above)
Comment (cconard96) [reply src/CommonITILObject.php:126]: No longer needed. 10.0 fix changes the `if` condition so `getActors` isn't called at all for new items.

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — COMMENTED** [review]
PHPStan issue has to be fixed.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**